src/linear_algebra/affine_space/ordered.lean

```diff
@@ -0,0 +1,278 @@
+end
+
+lemma line_map_mono_endpoints (ha : a ≤ a') (hb : b ≤ b') (h₀ : 0 ≤ r) (h₁ : r ≤ 1) :
+  line_map a b r ≤ line_map a' b' r :=
+(line_map_mono_left ha h₁).trans (line_map_mono_right hb h₀)
+
+lemma line_map_strict_mono_endpoints (ha : a < a') (hb : b < b') (h₀ : 0 ≤ r) (h₁ : r ≤ 1) :
+  line_map a b r < line_map a' b' r :=
+begin
+  rcases h₀.eq_or_lt with (rfl|h₀), { simpa },
+  exact (line_map_mono_left ha.le h₁).trans_lt (line_map_strict_mono_right hb h₀)
+end
+
+lemma line_map_lt_line_map_iff_of_lt (h : r < r') :
+  line_map a b r < line_map a b r' ↔ a < b :=
+begin
+  simp only [line_map_apply_module],
+  rw [← lt_sub_iff_add_lt, add_sub_assoc, ← sub_lt_iff_lt_add', ← sub_smul, ← sub_smul,
+    sub_sub_sub_cancel_left, smul_lt_smul_iff_of_pos (sub_pos.2 h)]
+end
+
+lemma left_lt_line_map_iff_lt (h : 0 < r) : a < line_map a b r ↔ a < b :=
+iff.trans (by rw line_map_apply_zero) (line_map_lt_line_map_iff_of_lt h)
+
+lemma line_map_lt_left_iff_lt (h : 0 < r) : line_map a b r < a ↔ b < a :=
+@left_lt_line_map_iff_lt k (order_dual E) _ _ _ _ _ _ h
+
+lemma line_map_lt_right_iff_lt (h : r < 1) : line_map a b r < b ↔ a < b :=
+iff.trans (by rw line_map_apply_one) (line_map_lt_line_map_iff_of_lt h)
+
+lemma right_lt_line_map_iff_lt (h : r < 1) : b < line_map a b r ↔ b < a :=
+@line_map_lt_right_iff_lt k (order_dual E) _ _ _ _ _ _ h
+
+end ordered_ring
+
+section linear_ordered_field
+
+variables [linear_ordered_field k] [ordered_add_comm_group E] [ordered_semimodule k E]
+
+section
+
+variables {a b : E} {r r' : k}
+
+lemma line_map_le_line_map_iff_of_lt (h : r < r') :
+  line_map a b r ≤ line_map a b r' ↔ a ≤ b :=
+begin
+  simp only [line_map_apply_module],
+  rw [← le_sub_iff_add_le, add_sub_assoc, ← sub_le_iff_le_add', ← sub_smul, ← sub_smul,
+    sub_sub_sub_cancel_left, smul_le_smul_iff_of_pos (sub_pos.2 h)]
+end
+
+lemma left_le_line_map_iff_le (h : 0 < r) : a ≤ line_map a b r ↔ a ≤ b :=
+iff.trans (by rw line_map_apply_zero) (line_map_le_line_map_iff_of_lt h)
+
+lemma line_map_le_left_iff_le (h : 0 < r) : line_map a b r ≤ a ↔ b ≤ a :=
+@left_le_line_map_iff_le k (order_dual E) _ _ _ _ _ _ h
+
+lemma line_map_le_right_iff_le (h : r < 1) : line_map a b r ≤ b ↔ a ≤ b :=
+iff.trans (by rw line_map_apply_one) (line_map_le_line_map_iff_of_lt h)
+
+lemma right_le_line_map_iff_le (h : r < 1) : b ≤ line_map a b r ↔ b ≤ a :=
+@line_map_le_right_iff_le k (order_dual E) _ _ _ _ _ _ h
+
+end
+
+/-!
+### Convexity and slope
+
+In this section we prove inequality that relate `f (line_map a b r) ≤ line_map (f a) (f b) r` and
+`f (line_map a b r) < line_map (f a) (f b) r` to inequalities on slopes of `f`. For each inequality
+we provide 3 lemmas:
+
+* `*_left` relates it to an inequality on `slope f a (line_map a b r)` and `slope f a b`;
+* `*_right` relates it to an inequality on `slope f a b` and `slope f (line_map a b r) b`;
+* no-suffix version relates it to an inequality on `slope f a (line_map a b r)` and
+  `slope f (line_map a b r) b`.
```

Comment:
Those lemmas are not fun to read. Maybe docstrings would help, maybe not...

Reply:
I added some docstrings.